.clang-tidy

```diff
@@ -3,19 +3,29 @@
 InheritParentConfig: true
 Checks: '
 bugprone-*,
+-bugprone-argument-comment,
+-bugprone-easily-swappable-parameters,
 -bugprone-forward-declaration-namespace,
 -bugprone-macro-parentheses,
+-bugprone-narrowing-conversions
 -bugprone-lambda-function-name,
 -bugprone-reserved-identifier,
+-bugprone-swapped-arguments,
+-bugprone-implicit-widening-of-multiplication-result,
 cppcoreguidelines-*,
+-cppcoreguidelines-avoid-c-arrays,
+-cppcoreguidelines-avoid-const-or-ref-data-members
+-cppcoreguidelines-avoid-do-while,
 -cppcoreguidelines-avoid-magic-numbers,
 -cppcoreguidelines-avoid-non-const-global-variables,
 -cppcoreguidelines-interfaces-global-init,
 -cppcoreguidelines-macro-usage,
+-cppcoreguidelines-narrowing-conversions,
```

Comment:
These are annoying, but we do have fixits for them, right? Why disable them?

Reply:
Many places are not suppressed, and they are very annoying when performing whole project scan. Narrowing conversions are unavoidable in a numerical library

Reply:
Yes, let's revert them